src/services/MobileService.ts

```diff
@@ -282,12 +276,16 @@ class MobileService {
     try {
       const exchangeUserId = Number(req.params['exchangeUserId'])
       const backupMethod = req.query.backupMethod || 'UNKNOWN'
+      const backupSharePairId = req.query.backupSharePairId
 
       const user = await this.getUserByExchangeId(exchangeUserId)
 
-      // Attempt to find the client backup share for the specified backup method.
+      // Attempt to find the client backup share by backupSharePairId or by backup method as a fallback.
       const clientBackupShare = user.clientBackupShares?.find(
-        (clientBackupShare) => clientBackupShare.backupMethod === backupMethod,
+        (clientBackupShare) =>
+          backupSharePairId
+            ? clientBackupShare.backupSharePairId === backupSharePairId
```

Comment:
If v3's behavior is going to have the customers store by backupSharePairId and retrieve by it, this makes sense. But if its not, then we should not add this in

Reply:
Blake is using it for v3 multi-wallet